### PR TITLE
Created algocfg to provide cli access to config.json

### PR DESCRIPTION
**Pull request by Karmastic**
Adds `algocfg get|set|reset` commands to manipulate config.json
in a consistent and typesafe way.
Modified config.json is persisted minimally - only non-default
values are written.  Versioning is taken into account and migration
is performed as required.

This also includes a fix to SaveNonDefaultValuesToFile() which was
writing config.json files with a trailing empty line for each default
value that was skipped.

### CODE REVIEW

File: cmd/algocfg/datadir.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+package main
+
+import "os"
+
+var dataDirs []string
+
+func resolveDataDir() string {
```
Comment (algobolson) [inline cmd/algocfg/datadir.go]: these functions exist elsewhere (cmd/goal/... at least) and maybe it's time to extract them? If not libgoal than some other library of algo-command-utils

File: cmd/algocfg/setCommand.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	err := setFieldValue(f, value)
+	return object, err
+}
+
+func setFieldValue(field reflect.Value, value string) error {
+	switch k := field.Kind(); k {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		val, err := strconv.ParseInt(value, 10, 64)
```
Comment (algobolson) [inline cmd/algocfg/setCommand.go]: we could ParseInt(value, 0, 64), 0-base allows for detecting 0777 octal or 0xabc hex. (Also ParseUint below)

File: cmd/algocfg/datadir.go (hunk 1 above)
Comment (Karmastic) [reply cmd/algocfg/datadir.go]: Yes they do.  I moved in the direction of extracting them in this PR - moving them to distinct files.  I didn't want to incur the extra cost right now.

File: cmd/algocfg/setCommand.go (hunk 2 above)
Comment (Karmastic) [reply cmd/algocfg/setCommand.go]: We could, but we do not want to or need to.  I don't want it to be ambiguous.

**Pull request merged**